service/storage_proxy.cc

```diff
@@ -481,10 +481,12 @@ class abstract_write_response_handler : public seastar::enable_shared_from_this<
     }
     // return true on last ack
     bool response(gms::inet_address from) {
-        auto it = _targets.find(from);
+        auto it = boost::find(_targets, from);
         if (it != _targets.end()) {
             signal(from);
-            _targets.erase(it);
+            using std::swap;
+            swap(*it, _targets.back());
```

Comment:
I'm curious, why "using std::swap" for just one use, instead of just writing explicitly std::swap as we always do?

Reply:
It's the correct way, see https://stackoverflow.com/questions/4782692/what-does-using-stdswap-inside-the-body-of-a-class-method-implementation-mea/4782809. Although I see it's really needed for templates, which this isn't.

Reply:
Interesting but sad.
Ok.